intuita-webview/src/jobDiffView/DiffViewer/Diff.tsx

```diff
@@ -1,14 +1,58 @@
-import MonacoDiffEditor from '../../shared/Snippet/DiffEditor';
+import { useRef, useState } from 'react';
+import MonacoDiffEditor, { monaco } from '../../shared/Snippet/DiffEditor';
 import { JobDiffViewProps } from '../App';
 
-export const DiffViewer = ({
+export type DiffChanges = { added: number; removed: number };
+
+export const useDiffViewer = ({
 	oldFileContent,
 	newFileContent,
 	viewType,
 }: JobDiffViewProps & { viewType: 'inline' | 'side-by-side' }) => {
-	return (
+	const editorRef = useRef<monaco.editor.IStandaloneDiffEditor | null>(null);
+	const [diffChanges, setDiffChanges] = useState<DiffChanges | null>(null);
+
+	const getDiffChanges = (): DiffChanges | undefined => {
+		if (!editorRef.current) {
+			return;
+		}
+		const lineChanges = editorRef.current.getLineChanges();
+		if (!lineChanges) {
+			return;
+		}
+
+		const addedAndDeletedCount = lineChanges.reduce(
```

Comment:
I would use `forEach` instead of reduce with `added` and `removed` outside of the callback for better readability.

Reply:
Perhaps you could move these calculations to a separate function.